File: Daily Challenge/February 2025/The k-th Lexicographical String of All Happy Strings of Length n.cpp
class Solution {
  public:
    string getHappyString(int n, int k) {
      string s = "", ans = "";
      backtrack(s, n, k, ans);
      
      return ans;
    }

  private:
    void backtrack(string &s, int n, int &k, string &ans) {
      if (s.size() == n) {
        if (--k == 0) {
          ans = s;
        }
        return;
      }

      for (char c = 'a'; c <= 'c'; ++c) {
        if (s.empty() || s.back() != c) {
          s.push_back(c);
          backtrack(s, n, k, ans);
          s.pop_back();
        }
      }
    }
  };

File: Daily Challenge/February 2025/The k-th Lexicographical String of All Happy Strings of Length n.cs
public class Solution {
  public string GetHappyString(int n, int k) {
    string current = "";
    Backtrack(n, ref k, current);

    return _result;
  }

  private string _result = "";

  private void Backtrack(int n, ref int k, string current) {
    if (current.Length == n) {
      k--;
      if (k == 0) {
        _result = current;
      }
    } else {
      for (char c = 'a'; c <= 'c'; c++) {
        if (current.Length == 0 || current[current.Length - 1] != c) {
          Backtrack(n, ref k, current + c);
        }
      }
    }
  }
}